### PR TITLE
Rename difficulty to totalDifficulty

### PR DESCRIPTION
**Pull request by dbrajovic**
# Description

Change varaible names and function calls where `totalDifficulty` is meant by `difficulty`

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

Thank you for improving the readability on this 🙏

**Review by Kourin1996 — APPROVED** [review]
LGTM. It's become more clear 🤗

File: blockchain/blockchain.go [hunk 1]
```diff
@@ -322,25 +322,25 @@ func (b *Blockchain) advanceHead(newHeader *types.Header) (*big.Int, error) {
...
 		}
-		currentDiff = td
+		parentTD = td
 	}
 
-	// Calculate the new difficulty
-	diff := big.NewInt(1).Add(currentDiff, new(big.Int).SetUint64(newHeader.Difficulty))
-	if err := b.db.WriteDiff(newHeader.Hash, diff); err != nil {
+	// Calculate the new total difficulty
+	newTD := big.NewInt(0).Add(parentTD, new(big.Int).SetUint64(newHeader.Difficulty))
```
Comment (brkomir) [inline blockchain/blockchain.go]: Maybe change big.NewInt(0) to new(big.Int) just for consistency

**Review by brkomir — APPROVED** [review]
I have left one comment but apart form that everything looks ok to me.

File: blockchain/blockchain.go (hunk 1 above)
Comment (dbrajovic) [reply blockchain/blockchain.go]: [Fixed](https://github.com/0xPolygon/polygon-sdk/pull/201/commits/5ee8d61e6fb83fde80026f0c01f558083e91d842)

I changed it to `big.NewInt(0)` to be consistent elsewhere within `blockchain.go`  at least.

**Review by lazartravica — APPROVED** [review]

**Pull request merged**